src/components/earn/ImportFarmModal.tsx

```diff
@@ -0,0 +1,169 @@
+import { getAddress } from '@ethersproject/address'
+import { ButtonError } from 'components/Button'
+import { LightCard } from 'components/Card'
+import { SearchInput } from 'components/SearchModal/styleds'
+import { useCustomStakingInfo } from 'pages/Earn/useCustomStakingInfo'
+import { FarmSummary } from 'pages/Earn/useFarmRegistry'
+import React, { RefObject, useContext, useEffect, useRef, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import { Text } from 'rebass'
+import styled, { ThemeContext } from 'styled-components'
+import { isAddress } from 'web3-utils'
+
+import { BIG_INT_SECONDS_IN_WEEK, IMPORTED_FARMS } from '../../constants'
+import { CloseIcon, TYPE } from '../../theme'
+import { AutoColumn } from '../Column'
+import Modal from '../Modal'
+import Row, { RowBetween } from '../Row'
+
+const ContentWrapper = styled(AutoColumn)`
+  width: 100%;
+  flex: 1 1;
+  position: relative;
+  padding: 1rem;
+`
+
+interface ImportFarmModalProps {
+  isOpen: boolean
+  onDismiss: () => void
+  farmSummaries: FarmSummary[]
+}
+
+export default function ImportFarmModal({ isOpen, onDismiss, farmSummaries }: ImportFarmModalProps) {
+  const { t } = useTranslation()
+  const inputRef = useRef<HTMLInputElement>()
+  const theme = useContext(ThemeContext)
+  const [farmAddress, setFarmAddress] = useState<string>('')
+  const { stakingToken, rewardTokens, totalRewardRates, valueOfTotalStakedAmountInCUSD } =
+    useCustomStakingInfo(farmAddress)
+  const [error, setError] = useState<string | undefined>(undefined)
+
+  const farmExists = isAddress(farmAddress)
+    ? farmSummaries.find(
+        (farmSummary) => farmAddress && getAddress(farmSummary.stakingAddress) === getAddress(farmAddress)
+      )
+    : undefined
+
+  const importedFarms = localStorage.getItem(IMPORTED_FARMS)
+
+  const handleInput = (event: any) => {
+    const input = event.target.value
+    setFarmAddress(input)
+  }
+
+  useEffect(() => {
+    if (isAddress(farmAddress)) {
+      const res = importedFarms
+        ? [...JSON.parse(importedFarms)].find((item) => getAddress(item) === getAddress(farmAddress))
```

Comment:
why are you doing `[...]`?

Why can't you just do `JSON.parse(importedFarms).find..`